libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -728,31 +733,28 @@ class MapboxNavigation(
         tripSession.updateSensorEvent(sensorEvent)
     }
 
-    private fun configureRouter() {
-        with(navigationOptions) {
-            // TODO StrictMode may report a violation as we're creating a File from the Main
-            val offlineFilesPath = OnboardRouterFiles(applicationContext, logger)
-                .absolutePath(onboardRouterOptions)
-            offlineFilesPath?.let { path ->
-                val routerParams = RouterParams(
-                    path,
-                    null,
-                    null,
-                    THREADS_COUNT,
-                    TileEndpointConfiguration(
-                        onboardRouterOptions.tilesUri.toString(),
-                        onboardRouterOptions.tilesVersion,
-                        accessToken ?: "",
-                        USER_AGENT,
-                        "",
-                        NativeSkuTokenProvider(
-                            MapboxNavigationAccounts.getInstance(applicationContext)
-                        )
+    private fun createTilesConfig(): TilesConfig {
+        // TODO StrictMode may report a violation as we're creating a File from the Main
+        val offlineFilesPath = OnboardRouterFiles(navigationOptions.applicationContext, logger)
+            .absolutePath(navigationOptions.onboardRouterOptions)
+        return offlineFilesPath?.let { path ->
+            return TilesConfig(
+                path,
+                null,
+                null,
+                THREADS_COUNT,
+                TileEndpointConfiguration(
+                    navigationOptions.onboardRouterOptions.tilesUri.toString(),
+                    navigationOptions.onboardRouterOptions.tilesVersion,
+                    navigationOptions.accessToken ?: "",
+                    USER_AGENT,
+                    "",
+                    NativeSkuTokenProvider(
+                        MapboxNavigationAccounts.getInstance(navigationOptions.applicationContext)
                     )
                 )
-                navigator.configureRouter(routerParams)
-            }
-        }
+            )
+        } ?: TilesConfig("", null, null, null, null)
```

Comment:
Log a warning/error or throw an exception instead?

Reply:
What about making `fun absolutePath(options: OnboardRouterOptions): String?` non-null and return `""` empty as we're already logging in there? That way we can remove `?:` altogether.

Reply:
Or maybe do:
```
val offlineFilesPath = OnboardRouterFiles(navigationOptions.applicationContext, logger)
    .absolutePath(navigationOptions.onboardRouterOptions) ?: ""
```

Reply:
I prefer the non-nullable option 😅

Reply:
:+1: